### PR TITLE
Adjust task executions group deletion

### LINKED ISSUE #1935: UI Group Delete of Task Executions causes deadlock
Issue:
The user attempts a group delete of task executions and the server responds with a 500 error stating that a deadlock has occurred. 
Cause:
A CTR definition that contains both boot3 and boot2 applications has been launched multiple times and has been restarted due to failures by a child task. When the user attempts to do a group delete of the task executions, it passes in a list of task executions to be removed without the associated schema type.   This causes deadlock issues when task execution and job execution data is being deleted.

Resolution:
Currently, the code https://github.com/spring-cloud/spring-cloud-dataflow-ui/blob/main/ui/src/app/shared/api/task.service.ts#L76-L78 Deletes all ids in one request.  
The solution to the issue is change the above implementation to one that does the following:

1. Create a list of the child task executions(task executions with a parent id) 
      a.  Group the child task execution ids by schemaTarget and invoke destroy with multiple ids for the schemaTarget
3.  Create a list of the parent task executions(task executions without a parent id) 
      a. Group by schemaTarget collect ids and invoke destroy with multiple ids for schemaTarget

Steps to reproduce:
1. Creates a composed task runner like `taskBoot2 && taskBoot3Fail && taskBoot3` as foo. (TaskBoot3Fail should fail)
2. Launch Foo
3. Once foo completes in a state of failed.   Go to the job execution page and restart.
4. Update taskBoot3Fail so that it will no longer fail. And compile to create a new binary or image
5. Go to the job execution page and restart
6. Upon its successful completion
7. Go to Task Execution Page and do a group delete all task executions.   

If you have any questions feel free to reach out to me or @corneil

### PR DESCRIPTION
**Pull request by claudiahub**
Resolves #1935

### CODE REVIEW

**Review by oodamien — APPROVED** [review]
LGTM

File: ui/src/app/shared/api/task.service.ts
Comment (corneil) [inline ui/src/app/shared/api/task.service.ts:1]: The groups of children will need to be completed before deleting the groups of parents.

Comment (corneil) [inline ui/src/app/shared/api/task.service.ts:1]: The groups of children will need to be completed before deleting the groups of parents.

Comment (corneil) [reply ui/src/app/shared/api/task.service.ts:1]:
```javascript
executionsClean(taskExecutions: TaskExecution[]): Observable<any> {
    const taskExecutionsChildren = taskExecutions.filter(taskExecution => taskExecution.parentExecutionId);
    const taskExecutionsParents = taskExecutions.filter(taskExecution => !taskExecution.parentExecutionId);
    return this.executionsCleanBySchema(taskExecutionsChildren).pipe(
      mergeMap(result => this.executionsCleanBySchema(taskExecutionsParents))
    );
  }

  executionsCleanBySchema(taskExecutions: TaskExecution[]): Observable<any> {
    const groupBySchemaTarget = taskExecutions.reduce((group, task) => {
      const schemaTarget = task.schemaTarget;
      group[schemaTarget] = group[schemaTarget] ?? [];
      group[schemaTarget].push(task);
      return group;
    }, {});
    const observables: Observable<any>[] = [];
    for (const schemaTarget in groupBySchemaTarget) {
      const group: TaskExecution[] = groupBySchemaTarget[schemaTarget];
      const ids = group.map(task => task.executionId);
      observables.push(this.taskExecutionsCleanByIds(ids, schemaTarget));
    }
    return forkJoin(observables);
  }
  taskExecutionsCleanByIds(ids: number[], schemaTarget: string): Observable<any> {
    const headers = HttpUtils.getDefaultHttpHeaders();
    const idStr = ids.join(',');
    const url =
      UrlUtilities.calculateBaseApiUrl() + `tasks/executions/${idStr}?action=CLEANUP,REMOVE_DATA&schemaTarget=${schemaTarget}`;
    return this.httpClient.delete<any>(url, {headers, observe: 'response'}).pipe(catchError(ErrorUtils.catchError));
  }
```

**Review by oodamien — APPROVED** [review]
LGTM

**Pull request merged**